Add ALO Kafka Connector Source

JONBRWN opened the pull request:
- add KafkaSourceReader, a Kafka Consumer with a resettable
    position
  - uses existing MultiSourceConnector with KafkaSourceReader
    in order to read from each partition in a topic

<!--
Make sure you've read the [Contributors Guide](https://github.com/WallarooLabs/wallaroo/blob/master/CONTRIBUTING.md). You'll need to sign our CLA before your issue can be accepted.

Reference the issue your code change relates to if possible
-->
closes #2989

<!--
Include any other necessary information below. If you have any questions don't hesitate to reach out on the mailing list or on IRC.
-->

jtfmumm commented on connectors/alo_kafka_source:
```diff
@@ -0,0 +1,87 @@
+	"""
+	A Kafka Consumer Source iterator with a resettable position.
+    """
+    :def __init__(self, topic, partition, bootstrap_servers):
```
A colon got in here: `:def`

JONBRWN replied:
fixed

jtfmumm commented:
This looks like a good start to get something working. Have you thought of a way to automate testing it?

JONBRWN commented:
@jtfmumm I've been thinking about it but am torn on how much value there is in adding the extra testing for this. The `MultiSourceConnector` is well tested as it's part of all of the ALO tests. This simply implements the `BaseSource` and `BaseIter` required for the `MultiSourceConnector`. 

Any thoughts?

jtfmumm commented:
@JONBRWN My thinking is that if we use this in one or more apps, we don't want it to break independently of the `MultiSourceConnector` code. Do you have an idea of what it would take to set up an example app that does the end to end integration? Another option is to mark this code as experimental since there's a good chance we won't ultimately use a Python version of the Kafka connector and it's primary purpose at the moment is to get some exploratory manual tests working.

JONBRWN commented:
@jtfmumm I think marking this as experimental is the best option. I don't imagine us using this outside of quickly iterating on an application and making sure the Kafka source related components of that application work. The time it would take to fix any part of this in the event it breaks is far smaller than the time it would take to set up automated testing for it.

slfritchie commented:
My thought on this is to adapt the current suite of correctness tests: 1. put its source data into Kafka, and 2. read its sink data from Kafka.  That also feels like a big project for an experimental feature.

JONBRWN commented:
@slfritchie that was my initial thought as well. there are however a decent amount of unknowns in setting up Kafka (either via Docker or other means) both in and out of the CI environment to get this type of test to work effectively.

JONBRWN commented:
@jtfmumm I moved the kafka source to an experimental folder and left a note in the documentation in the file. I didn't update the actual Wallaroo documentation due to the fact that it hasn't been updated at all for the new connectors. 

I've removed the `DO NOT MERGE` label, feel free to merge if you don't have any more questions.

jtfmumm commented:
@JONBRWN Changes look good.  Can you check out these failing Travis tests? If it's spurious, this can be merged.

JONBRWN commented:
@jtfmumm the failing Travis tests are unrelated and due to a change in the environment, merging
